core/Archive/ArchivePurger.php

```diff
@@ -154,6 +156,84 @@ public function purgeOutdatedArchives(Date $dateStart)
         return $deletedRowCount;
     }
 
+    public function purgeDeletedSiteArchives(Date $dateStart)
+    {
+        $idArchivesToDelete = $this->getDeletedSiteArchiveIds($dateStart);
+
+        $deletedRowCount = 0;
+        if (!empty($idArchivesToDelete)) {
+            $deletedRowCount = $this->deleteArchiveIds($dateStart, $idArchivesToDelete);
+
+            $this->logger->info("Deleted {count} rows in archive tables (numeric + blob) for deleted sites for {date}.", array(
+                'count' => $deletedRowCount,
+                'date' => $dateStart
+            ));
+
+            $this->logger->debug("[Deleted IDs: {deletedIds}]", array(
+                'deletedIds' => implode(',', $idArchivesToDelete)
+            ));
+        } else {
+            $this->logger->debug("No archives for deleted sites found in archive numeric table for {date}.", array('date' => $dateStart));
+        }
+
+        return $deletedRowCount;
+    }
+
+    public function purgeDeletedSegmentArchives(Date $dateStart, array $validSegmentHashes)
+    {
+        $idArchivesToDelete = $this->getDeletedSegmentArchives($dateStart, $validSegmentHashes);
+
+        $deletedRowCount = 0;
+        if (!empty($idArchivesToDelete)) {
+            $deletedRowCount = $this->deleteArchiveIds($dateStart, $idArchivesToDelete);
+
+            $this->logger->info("Deleted {count} rows in archive tables (numeric + blob) for deleted segments for {date}.", array(
+                'count' => $deletedRowCount,
+                'date' => $dateStart
+            ));
+
+            $this->logger->debug("[Deleted IDs: {deletedIds}]", array(
+                'deletedIds' => implode(',', $idArchivesToDelete)
+            ));
+        } else {
+            $this->logger->debug("No archives for deleted segments found in archive numeric table for {date}.", array('date' => $dateStart));
+        }
+
+        return $deletedRowCount;
+    }
```

Comment:
There's some code redundancy in purgeDeletedSiteArchives & purgeDeletedSegmentArchives which could be removed.

Reply:
Yup, have moved this logic into a purge() method